custom/icds_reports/filters.py

```diff
@@ -35,40 +35,66 @@ def loc_to_json(loc, project):
             'can_edit': True
         }
 
-    project = Domain.get_by_name(domain)
+    def _get_accessible_location_ids(primary_location):
+        accessible_location_ids = set([primary_location.location_id])
+        accessible_location_ids.update(primary_location.get_ancestors().values_list('location_id', flat=True))
+        accessible_location_ids.update(primary_location.get_descendants().values_list('location_id', flat=True))
+        return accessible_location_ids
+
+    def _get_ancestor_loc_dict(location_list, location):
+        for parent_location in location_list:
+            if parent_location['uuid'] == location.location_id:
+                return parent_location
+        return None
+
+    user_location = user.get_sql_location(domain)
```

Comment:
hmm, I thought we wanted to just check for the primary location of the user. Will let @Rohit25negi share why he used that. That was the one reason we didn't end up using `accessible_to_user` method because that uses assigned location and not primary location and ended up finding out the accessible locations ourselves.

Reply:
Now i remember, I think i got to believe that because of https://github.com/dimagi/commcare-hq/pull/28290/files#r467858278

Reply:
oh that's a good point. Do you think its worth doing that? because On a real basis it never happens I think and national level user anyway has national-level access which we are already handling it here.?

Reply:
Nvm, I think it's worth adding that support in case. Done making that change.